src/support.cpp

```diff
@@ -783,6 +788,213 @@ void AreaSupport::generateOverhangAreasForMesh(SliceDataStorage& storage, SliceM
                                });
 }
 
+Polygons AreaSupport::generateVaryingXYDisallowedArea(const SliceMeshStorage& storage, const Settings& infill_settings, const LayerIndex layer_idx)
+{
+    const auto& mesh_group_settings = Application::getInstance().current_slice->scene.current_mesh_group->settings;
+    const auto layer_thickness = mesh_group_settings.get<coord_t>("layer_height");
+    const auto support_distance_top = mesh_group_settings.get<coord_t>("support_top_distance");
+    const auto support_distance_bot = mesh_group_settings.get<coord_t>("support_bottom_distance");
+    const auto overhang_angle = mesh_group_settings.get<AngleRadians>("support_angle");
+    const auto xy_distance = static_cast<double>(mesh_group_settings.get<coord_t>("support_xy_distance"));
+
+    constexpr coord_t snap_radius = 10;
+    constexpr coord_t close_dist = 15; // needs to be larger than the snap radius!
+    constexpr coord_t search_radius = 0;
+
+    auto layer_current = storage.layers[layer_idx].getOutlines().offset(-close_dist).offset(close_dist).smooth(close_dist);
+
+    // sparse grid for storing the offset distances at each point. For each point there can be multiple offset
+    // values as multiple may be calculated when multiple layers are used for z-smoothing of the offsets.
+    // The average of all offset dists is taken for the used varying offset. To account for this the commutative
+    // offset, and the number of offsets $n$ are stored simultaneously. The final offset used is then commutative
+    // equal to commutative_offset / n.
+    SparsePointGridInclusive<std::pair<size_t, double>> offset_dist_at_point(snap_radius);
+
+    // Collection of the various areas we used to calculate the areas for. This is a combination
+    //  - the support distance (this is the support top distance for overhang areas, and support
+    //    bottom thickness for sloped areas)
+    //  - of the delta z between the current layer and layer below (this can vary between the areas
+    //    when we use multiple layers for z-smoothing)
+    //  - the polygon delta; the xy-distance is calculated separately for overhang and sloped areas.
+    //    here either the slope or overhang area is stored
+    std::vector<std::tuple<double, double, Polygons>> z_distances_layer_deltas;
+
+    constexpr LayerIndex layer_index_offset = 1;
+
+    const LayerIndex layer_idx_below = std::max(layer_idx - layer_index_offset, static_cast<LayerIndex>(0));
+    if (layer_idx_below != layer_idx)
+    {
+        auto layer_below = storage.layers[layer_idx_below].getOutlines()
+                               .offset(-close_dist)
+                               .offset(close_dist)
+                               .smooth(close_dist);
+
+        z_distances_layer_deltas.emplace_back(
+            static_cast<double>(support_distance_top),
```

Comment:
```suggestion
            support_distance_top,
```

see earlier cast above